lstchain/image/muon/muon_analysis.py

```diff
@@ -87,19 +91,15 @@ def fit_muon(x, y, image, geom, tailcuts):
     image_clean = image * clean_mask
     muonringparam = muonring.fit(x, y, image_clean)
 
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2)
-                       + np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
-        )
-
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2) +
-                       np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
+    # Do an iterative fit removing pixels which are beyond 0.4*ring_radius of the ring (along the radial direction):
+    # The goal is to improve fit for good rings with very few additional non-ring bright pixels.
 
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
+    for _ in [0]*2:  # just to iterate the fit twice more
+        dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2)
```

Comment:
use the power operator, not the power function: `np.sqrt((x - c_x)**2 + (y - c_y)**2)`, see discussion in https://github.com/cta-observatory/ctapipe/issues/1028

Reply:
Done, thx